src/software/ai/hl/stp/tactic/shadow_enemy_tactic.h

```diff
@@ -32,20 +32,27 @@ class ShadowEnemyTactic : public Tactic
     std::string getName() const override;
 
     /**
-     * Updates the parameters for this ShadowEnemyTactic.
+     * Updates the world parameters for this ShadowEnemyTactic.
      *
      * @param enemy_threat The EnemyThreat indicating which enemy to shadow
      * @param field The field being played on
      * @param friendly_team The friendly team
      * @param enemy_team The enemy team
-     * @param enemy_team_can_pass Whether or not we think the enemy team can pass the ball
      * @param shadow_distance How far from the enemy the robot will shadow. This is the
      * distance between the center of the enemy robot and the center of the robot
      * shadowing it
      */
-    void updateParams(const Evaluation::EnemyThreat &enemy_threat, const Field &field,
-                      const Team &friendly_team, const Team &enemy_team,
-                      double shadow_distance, bool enemy_team_can_pass, const Ball &ball);
+    void updateWorldParams(const Evaluation::EnemyThreat &enemy_threat,
+                           const Field &field, const Team &friendly_team,
+                           const Team &enemy_team, double shadow_distance,
+                           const Ball &ball);
+
+    /**
+     * Updates the control parameters for this ShadowEnemyTactic
+     *
+     * @param enemy_team_can_pass Whether or not we think the enemy team can pass the ball
+     */
+    void updateControlParams(bool enemy_team_can_pass);
```

Comment:
Addressed in feb60a45ab8b5243f2d0a31f8211a16a74181ea6 and 70a8bf4fe93a5a35f04fee4ff31010ce0a7775ea. While I was moving `enemy_team_can_pass`, I noticed that the parameter `ball_steal_speed` is never really assigned; in the constructor, the argument gets passed in but is never assigned to the parameter, and in `updateWorldParams`, it was not given as an argument so it was being assigned to itself (I removed it in 51865437d9b10cbde18e295a7fdbd7ef6338b871 since it wasn't doing anything). Should this parameter be removed from the class?

Reply:
Gut reaction is yes, but @MathewMacDougall can you confirm since you wrote this tactic?

Reply:
You're right that 'ball_steal_speed` wasn't assigned in the constructor, whoopes :sweat_smile:. I think that it can just be put in the constructor and removed from the `update...` functions since I can't think of a use case where it would change mid-Play, but it is being used in the `calculateNextIntent` function.

Reply:
Assigning `ball_steal_speed` in the constructor seems to break `shadow_enemy_tactic_test`, so I think I want to create a separate issue to fix this.

Reply:
Ok that's fine